### PR TITLE
Bulk Upload Domains UI

### LINKED ISSUE #100: Bulk add domains
# 💡 Summary #

Add a list of domains in bulk

## Motivation and context ##

In some cases there will be hundreds of domains that need to be added at a time.

This would be useful because...

Minimize the time it takes to add domains to Domain Manager

## Acceptance criteria ##

How do we know when this work is done?

will update issue to resolved once the feature has been created.

- [ ] Criterion

### PR DESCRIPTION
**Pull request by dylanj1752**
## 💭 Description

Ability to upload bulk domains into the ui. Input box altered to allow for extended input.

Takes in a comma separated list (can be new line each entry but does not have to be). Filters each entry to ensure it is a valid url, alerting the user of any that do not meet the url format. 

If all are valid, submits to api and adds to the domain list. (Tested against DM-API branch dm-308)

## ✅ Checklist

- [ x ] My code follows the code style of this project.
- [ x ] My changes have been adequately tested locally.
- [ ] All automated tests are passing.
- [ ] I have updated/added required automated tests.
- [ x ] Pre-commit checks are passing.

### CODE REVIEW

File: src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html [hunk 1]
```diff
@@ -1,38 +1,68 @@
...
-<form [formGroup]="domainForm" class="dialog text-muted" mat-dialog-content>
+<form
+  [formGroup]="domainForm"
+  class="dialog text-muted"
+  style="height: 25em"
+  mat-dialog-content
+>
   <div class="text-muted mb-4">
-    Provide a domain and select submit to create a new domain
+    Please enter one or more domains to add, seperated by a comma.
```
Comment (zenine07) [inline src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html]: I would switch this to `Please enter one or more domains separated by new lines`

File: src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html [hunk 2]
```diff
@@ -1,38 +1,68 @@
...
-        <input
+        <mat-label>Domain URL's</mat-label>
+        <textarea
           matInput
           trim="blur"
           type="text"
           formControlName="url"
           placeholder=" The url of the new domain"
-        />
+        ></textarea>
```
Comment (zenine07) [inline src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html]: Can we get this to expand a bit more by default? Doesn't look that great on my screen.

image

File: src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html [hunk 3]
```diff
@@ -1,38 +1,68 @@
...
+    class="mb-3"
+    style="margin-right: auto"
+    (click)="openFileBrowser($event)"
+  >
+    Upload CSV List <i class="fa fa-upload"></i>
+  </button>
+  <input
+    id="csvUpload"
+    type="file"
+    accept=".csv,.xls,.xlsx"
```
Comment (zenine07) [inline src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html]: I'd accept .txt files as well, like we had discussed earlier.

**Review by zenine07 — CHANGES_REQUESTED** [review]
Did not work for me. I put a domain per line like this

```
zachtest1.xyz
zachtest2.xyz
zachtest3.xyz
```

And the resulting domain was `zachtest1.xyzzachtest2.xyzzachtest3.xyz`

We shouldn't need to separate domains by commas, just new lines as we just need a list of domains to create.

File: src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html (hunk 1 above)
Comment (dylanj1752) [reply src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html]: Modified

File: src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html (hunk 2 above)
Comment (dylanj1752) [reply src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html]: Expanded more by default.

File: src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html (hunk 3 above)
Comment (dylanj1752) [reply src/domainManagementUI/src/app/components/domain/domain-create-dialog/domain-create-dialog.component.html]: allows txt files

**Review by zenine07 — APPROVED** [review]
Tested great!

**Pull request merged**